app/models/user.rb

```diff
@@ -24,18 +24,23 @@ def pending_friends
   end
 
   # Users who have requested to be friends
+  # also doubles as pending_requests
   def friend_requests
-    inverse_friendships.map { |friendship| friendship.user unless friendship.confirmed }.compact
+    # friendships.map { |friendship| friendship unless friendship.confirmed }.compact
+    # inverse_friendships.map { |friendship| friendship.user unless friendship.confirmed }.compact
+    Friendship.all.where(friend_id: id, confirmed: false)
   end
 
   def confirm_friend(user)
```

Comment:
```
has_many :confirmed_friendships, -> { where confirmed: true }, class_name: "Friendship"
has_many :friends, through: :confirmed_friendships
```

- [ ] confirm_friend can be defined as association with SQL query instead of processing objects in application memory (as the database was created to store and query data it is always more efficient than ruby)